distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -836,29 +911,22 @@ type File_Error
 
        Arguments:
        - file: The file that doesn't exist.
-    type No_Such_File_Error file
-
-    ## An error that indicates that the program does not have access to the
-       requested file.
-
-       Arguments:
-       - file: The file that the program does not have permission to access.
-    type Access_Denied_Error file
+    type File_Not_Found file
 
     ## A generic IO error.
 
        Arguments:
+       - file: The file that couldn't be read.
        - message: The message for the error.
-    type Io_Error message
+    type Io_Error file message
 
     ## UNSTABLE
 
        Convert the File error to a human-readable format.
     to_display_text : Text
     to_display_text = case this of
-        No_Such_File_Error file -> "The file at " + file.path + " does not exist."
-        Access_Denied_Error file -> "You do not have permission to access the file at " + file.path + "."
-        Io_Error msg -> "An IO error has occurred: " + msg.to_text + "."
+        File_Not_Found  file -> "The file at " + file.path + " does not exist."
+        Io_Error file msg -> msg.to_text + " (" + file.path + ")."
```

Comment:
Now I wonder though if the Io_Error coming from `handle_java_exceptions` will not contain a trailing dot coming from `exc.getMessage`? That's why I was unsure if it is feasible to handle this dot correctly.

Reply:
Will leave as is - if we encounter this as an issue can do a trim `.` if needed